ts/sagas/wallet/pagopaApis.ts

```diff
@@ -59,20 +60,29 @@ export function* fetchWalletsRequestHandler(
   pagoPaClient: PaymentManagerClient,
   pmSessionManager: SessionManager<PaymentManagerToken>
 ): Iterator<Effect> {
-  const request = pmSessionManager.withRefresh(pagoPaClient.getWallets);
-  try {
-    const getResponse: SagaCallReturnType<typeof request> = yield call(request);
-    if (getResponse.isRight()) {
-      if (getResponse.value.status === 200) {
-        yield put(fetchWalletsSuccess(getResponse.value.value.data));
+  yield put(checkCurrentSession.request());
+  const sessionAction: ActionType<
+    typeof checkCurrentSession["success"]
+  > = yield take(checkCurrentSession.success);
```

Comment:
Here we could have a lock: we are waiting for `checkCurrentSession.success` action but it could happen it fails (`checkCurrentSession.failure`)
In that case the execution will be blocked at that point.

I guess we could dispatch the check action
`yield put(checkCurrentSession.request());` and do nothing more.
If the session is invalid a sessionExpired will be dispatched and the logout is forced.

Do you find weak points on this basic logic?

Reply:
Sounds good!
Fixed in db9ddc7